sql/sql_parse.cc

```diff
@@ -9309,16 +9325,30 @@ kill_one_thread(THD *thd, my_thread_id id, killed_state kill_signal, killed_type
         WSREP_DEBUG("kill_one_thread victim: %lld wsrep_aborter %lu"
                     " by signal %d",
                     id, tmp->wsrep_aborter, kill_signal);
+
+#ifdef WITH_WSREP
+        if (WSREP(thd))
+        {
+          wsrep_abort_thd(thd, tmp, 1);
```

Comment:
If `wsrep_abort_thd()` and methods called from it release `LOCK_thd_kill`, it is unsafe to access thd object after this line. This applies also to ` if (!WSREP(thd) || !wsrep_abort_used)` conditions below. Instead of relying `wsrep_abort_used` flag, maybe better approach would be early return here.

Reply:
thd is the BF thread and the victim is tmp, which was locked by LOCK_thd_kill prior to wsrep_abort_thd() call. tmp is not used after this point.
But early return after wsrep_abort_thd(), is gives better look for the code, will change this

Reply:
Got confused between `thd` and `tmp`. But this raises another question, why the if condition is for `WSREP(thd)`, not `WSREP(tmp)`. 

I also see that this  PR skips unlocking `tmp->LOCK_thd_kill` in this function, but I can't find any other place in this PR where the `tmp->LOCK_thd_kill` would be unlocked. This would leave `tmp->LOCK_thd_kill` locked after leaving this function, and could produce a deadlock later on.

Reply:
mysql_mutex_unlock(&tmp->LOCK_thd_kill); is called quite at the end of this function

Reply:
the locking policy for wsrep_abort_thd() is entered with LOCK_thd_data and LOCK_thd_kill locked, and these mutexes are unlocked at return. 
The early return is now right after the wsrep_abort_thd()  call, so tmp should be in totally unlocked state then